services/core-web/src/components/common/DocumentTable.tsx

```diff
@@ -93,8 +123,17 @@ export const DocumentTable = ({
       return doc;
     });
   };
+
   const documents = parseDocuments(props.documents ?? []);
 
+  useEffect(() => {
+    const isMultiSelect = documents.every((doc) =>
```

Comment:
Can we use the enableBulkActions prop? Can get rid of a state variable that way.

Reply:
Initially I tried setting up the condition check to see if every document in the table has permission within the MajorMineApplicationTab.js file, and the boolean result of that check would be what gets passed into enableBulkAcations prop.. however I ran into an issue where in MajorMineApplicationTab.js's  renderDocumentSection function you console.log the sectionDocuments, you would see documents having 6 allowed_actions, but if you were to access the same document's allowed_actions, for some reason it only shows 2 allowed_actions "Open in document viewer" and "Download File". Not too sure what was going on there
![Screenshot (155)](https://github.com/bcgov/mds/assets/127789479/1a251181-727a-46ff-9a67-b16374a2be86)

However I think it makes sense to separate enableBulkAction prop from the check to see if all documents have permission, because technically even if the documents don't have the major mine application edit permission, the user should still be able to bulk download, just won't have permission to bulk archive.

Reply:
Let me know if you're still having trouble with this and we can figure it out over VC. I had issues getting the permissions _as the result of a function_ (because the table will, in certain circumstances, turn an object proper into a plain JSON object with no methods) but there should not be any issues getting its attributes and we should sort that out. Ideally we'd want to be able to do something like "multiFileActions = allowedTableActions.filter(action => documents.every(document has permission for that action))"

Reply:
As discussed, this issue seems to be happening outside of the document table and doesn't interfere with any logic, so should be okay to leave as it is.